interfaces/IDRM.h

```diff
@@ -95,15 +89,15 @@ class BufferReader {
 
     // Read a value from the stream, performing endian correction,
     // and advance the stream pointer.
-    inline bool Read1(uint8_t* v) WARN_UNUSED_RESULT { return Read(v); }
-    inline bool Read2(uint16_t* v) WARN_UNUSED_RESULT { return Read(v); }
-    inline bool Read2s(int16_t* v) WARN_UNUSED_RESULT { return Read(v); }
-    inline bool Read4(uint32_t* v) WARN_UNUSED_RESULT { return Read(v); }
-    inline bool Read4s(int32_t* v) WARN_UNUSED_RESULT { return Read(v); }
-    inline bool Read8(uint64_t* v) WARN_UNUSED_RESULT { return Read(v); }
-    inline bool Read8s(int64_t* v) WARN_UNUSED_RESULT { return Read(v); }
-
-    inline bool ReadString(std::string* str, size_t count) WARN_UNUSED_RESULT
+    inline bool Read1(uint8_t* v) WARNING_RESULT_NOT_USED { return Read(v); }
```

Comment:
Ahh this is the reason why you started using Thunder. Suggest to not use CORE, but do rename the preprocessor stuff to align with thunder. Just check if it is already defined, if not define it :-) but try to break the link with Thudner.

Reply:
Added Portability.h to support all these MACROS in interface scope